### DISCUSSION TITLE
preventing editor from scrolling when a key is pressed

### DISCUSSION THREAD

**Discussion by Raj-divine** [ajaxorg/ace#5092] [Q&A]
Hi everyone, I want to prevent the default behavior of the editor where if the cursor is not in view and you press a key it scrolls to the position of the cursor for some reason this is causing the text above the cursor to disappear and the cursor's behavior also changes instead of being on the line I clicked it jumps to some line above it

Because of these two reasons, I want to implement my own logic for scrolling and stop the editor's implementation.

I want to know where the logic for this scrolling is in the ace.js file so that I can remove it, or how can I prevent this behavior by any other method

thanks

**Comment by nightwing** [comment]
Editor's implementation is at https://github.com/ajaxorg/ace/blob/a9819722cec6ff60b028deaa7b70d7fefabac531/src/virtual_renderer.js#L1223, however stopping it is unlikely to produce good result, as it seems you have found a serious bug in the editor. 
If you could give us an example page where this issue can be reproduced, we could fix the bug which would save you and other users of ace lots of time.
Also it is worth to check that you are calling `editor.resize()` after changing its size.
